src/tools/toolbase.cmake

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function(add_quic_tool)
+    set(targetname ${ARGV0})
+    list(REMOVE_AT ARGV 0)
+
+    add_executable(${targetname} ${ARGV})
+    target_link_libraries(${targetname} msquic platform inc)
+    set_property(TARGET ${targetname} PROPERTY FOLDER "tools")
+endfunction()
+
+function(quic_tool_warnings)
+    target_link_libraries(${ARGV0} warnings)
+endfunction()
```

Comment:
Personally, I'm not sold on the necessity to even have this or link in warnings for any tools; just for simplicity sake.

Reply:
Some tools already have warnings enabled, about 1/3 of them. Wanted to keep the status quo. I would prefer any new tools have warnings enabled, because it really is better coding practice.